src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -1128,11 +1128,11 @@ private async Task EntityMiddleware(DispatchMiddlewareContext dispatchContext, F
                 }
             }
 
-            await entityContext.RunDeferredTasks();
-
-            // If there were internal errors, do not commit the batch, but instead rethrow
+            // If there were internal errors, throw a SessionAbortedException
             // here so DTFx can abort the batch and back off the work item
-            entityContext.ThrowInternalExceptionIfAny();

```

Comment:
I noticed that there were two places where we previously called `ThrowInternalExceptionIfAny`. Does the other location not require any changes?

Reply:
I think the other location is alright: it throws *after* DTFx has done its thing (persist history etc). At that point, throwing an exception is simply a way to tell the Functions telemetry that something went wrong inside this function. We do this for user code errors also, so they show up in the portal.

Reply:
I should probably test that.

Reply:
Ok, I tested it: it is fine. Those exceptions are being caught and wrapped, and then reported. My explanation above was not right though: it does not happen after DTFx finishes, but before.